lib/ssh_scan/scan_engine.rb

```diff
@@ -9,11 +9,34 @@ def scan_target(socket, opts)
       target, port = socket.chomp.split(':')
       policy = opts[:policy_file]
       timeout = opts[:timeout]
+      result = []
 
-      client = SSHScan::Client.new(target, port, timeout)
-      client.connect()
-      result = client.get_kex_result()
-      return result if result.include?(:error)
+      if target.fqdn?
+        if target.resolve_fqdn_as_ipv6.nil?
+          client = SSHScan::Client.new(target.resolve_fqdn_as_ipv4.to_s, port, timeout)
+          client.connect()
+          result = client.get_kex_result()
+          result[:hostname] = target
+          return result if result.include?(:error)
+        else
+          client = SSHScan::Client.new(target.resolve_fqdn_as_ipv6.to_s, port, timeout)
```

Comment:
@claudijd That is what exactly happening now I am first looking for AAAA records and if I get nil value, then I look for A records for IPv4 and in the last if I handle the case when IPv6 does not support connection to port then I go for IPv4 connection